### ISSUE TITLE
Fix version mismatch on immutable IAVL tree  in cosmos-sdk v0.45.12

### ISSUE THREAD

**Issue by danbryan** [noble-assets/noble#256]
cosmos-sdk v0.45.12 broke support for historical query of data and decided not to fix or accept others patches.  This is resolved in cosmos-sdk 47/50, but many chains still use 45.  @Reecepbcups has provided a patch for this issue.

https://github.com/Reecepbcups/cosmos-sdk/releases/tag/v0.45.16-fast.1

I will be submitting a PR to apply this patch.

**Comment by jonathanpberger** [comment]
Fixed in 46 and 47, but another issue came up. GRPC is acting up. This only affects archive nodes. Needs further sleuthing.

**Comment by jonathanpberger** [comment]
Waiting on Noble to decide whether they want to merge PR. Coding is done.

**Comment by jonathanpberger** [comment]
@boojamya please attach PR 257 and close out this ticket.

**Comment by akc2267** [comment]
up to Noble team at this point

**Comment by johnletey** [comment]
Closing this. Please see https://github.com/noble-assets/noble/pull/257#issuecomment-2140121216.
